### PR TITLE
Update to Calico v3.22.2

### PR DESCRIPTION
**Pull request by ScheererJ**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
Update to Calico v3.22.2

**Which issue(s) this PR fixes**:
None.

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

**Review by DockToFuture — APPROVED** [review]
/lgtm

File: charts/internal/calico/templates/custom-resource-definition/crd-ipreservations.yaml [hunk 1]
```diff
@@ -2,6 +2,9 @@
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
+  annotations:
```
Comment (DockToFuture) [inline charts/internal/calico/templates/custom-resource-definition/crd-ipreservations.yaml:5]: We can remote this annotation.

Comment (ScheererJ) [reply charts/internal/calico/templates/custom-resource-definition/crd-ipreservations.yaml]: Done.

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Pull request merged**